### DISCUSSION TITLE
Does the Android emulator still need to be installed just to build an APK?

### DISCUSSION THREAD

**Discussion by rmartin16** [beeware/briefcase#1366] [Q&A] [answered]
To build an APK, the Android emulator is installed; this was counterintuitive to me since the emulator obviously isn't necessary to build an APK but the note below from `gradle.py` suggests why this is:

```python
    def run_gradle(self, app, args):
        # Gradle may install the emulator via the dependency chain build-tools > tools >
        # emulator. (The `tools` package only shows up in sdkmanager if you pass
        # `--include_obsolete`.) However, the old sdkmanager built into Android Gradle
        # plugin 4.2 doesn't know about macOS on ARM, so it'll install an x86_64 emulator
        # which won't work with ARM system images.
        #
        # Work around this by pre-installing the emulator with our own sdkmanager before
        # running Gradle. For simplicity, we do this on all platforms, since the user will
        # almost certainly want an emulator soon enough.
        self.tools.android_sdk.verify_emulator()
```

Given Briefcase has [moved past](https://github.com/beeware/briefcase/pull/1271) Gradle 4.2 to Gradle 8.0, **is it still necessary to install the emulator before running Gradle?**

Note:
- The impetus for seems to have been macOS on arm64...but I don't have such a device to test this myself.
- On certain platforms, namely Linux on arm, the emulator installed by `sdkmanager` isn't particularly usable....and the emulator doesn't really run that well anyway on slow arm devices. These users would likely omit using the emulator altogether and just run on a real device which provides a much better experience.

**Comment by mhsmith** [comment, marked as answer]
Yes, I think this line can be removed now. If you want to make a PR doing that, I can test it on my arm64 Mac.

**Reply by rmartin16** [reply]
Thanks; I may do this.

However, in my testing, it doesn't seem to actually address my underlying issue. Even if Briefcase doesn't install the emulator itself here, Gradle will when it attempts to install the build tools....and on several systems, this results in an error. And then even if you fake out everything so it _appears_ as though the emulator is installed, Briefcase tries to invoke the emulator for `briefcase run android` to get a list of available AVDs.

This all is mostly in pursuit of trying to build and run an Android APK on a physical Android device from a host system where the emulator cannot be installed easily. I haven't looked in to it too deeply yet....but I guess this is all in vain if the emulator dependency cannot be omitted for the build-tools install.

**Reply by mhsmith** [reply]
Yes, I don't know why that dependency exists. 

On the other hand, as you mentioned in #1360, the Android development tools are only fully supported on 4 platforms: Linux on x86_64, Windows on x86_64, and Mac on x86_64 and ARM64. So even if we could avoid the emulator installation on Linux ARM, would we be able to get compatible builds of ADB and the other non-Java tools?

**Reply by rmartin16** [reply]
Indeed; that's the next hurdle.

Interestingly, if you fake an existing emulator installation on one of these unsupported platforms and then let `sdkmanager` try to install the other build tools, they'll all be for `x86_64`....so, its a mess all around.

That notwithstanding, my goal is to allow users to set up their own Android SDK on these platforms and let Briefcase play nicely with it. As long as the emulator install check is required, this is really only possible on Linux for `aarch64`....since building the emulator yourself for an arbitrary platform would likely be a big lift.

As for obtaining compatible versions of the tools, various places on the internet host versions for download...and they aren't _too_ bad to just compile yourself apparently. On Windows for ARM, there's support for x86 emulation out of the box....so, I'm hoping the x86_64 build tools would work. (I'm still working on setting up a bare-metal arm64 Windows install so I can see how Windows Subsystem for Android works with Briefcase).

At any rate, I am considering adding support for Briefcase to operate without the emulator....this would at least allow faking an emulator install if someone wanted to. On principle, at least, I think there's some standing for decoupling Briefcase from the emulator anyway.
